src/SAWScript/Interpreter.hs

```diff
@@ -1913,6 +1921,14 @@ primitives = Map.fromList
     , "The specified size must be greater than the size of the LLVM type."
     ]
 
+  , prim "crucible_array_alloc" "Int -> Term -> CrucibleSetup SetupValue"
```

Comment:
any suggestions regarding the name? I am also considering `crucible_raw_alloc`.

Reply:
Maybe `crucible_symbolic_alloc`? The fact that it allows a symbolic size seems like the key thing.

Reply:
done